core/domain/topic_domain.py

```diff
@@ -1184,6 +1193,29 @@ def validate(self, strict: bool = False) -> None:
                         'Subtopic with title %s does not have any skills '
                         'linked.' % subtopic.title)
 
+        all_skill_ids = self.get_all_skill_ids()
```

Comment:
Here I have considered all the skills which are assigned to a topic, i.e., (uncategorized skills + skills in subtopics).